src/Utilities/Compiler/Extensions/IOperationExtensions.cs

```diff
@@ -638,6 +638,38 @@ public static IOperation WalkDownConversion(this IOperation operation)
 
             return thrownObject?.Type;
         }
+
+        public static bool HasAnyExplicitDescendant(this IOperation operation, Func<IOperation, bool>? descendIntoOperation = null)
+        {
+            var stack = ArrayBuilder<IEnumerator<IOperation>>.GetInstance();
+            stack.Add(operation.Children.GetEnumerator());
+
+            while (stack.Any())
+            {
+                var enumerator = stack.Last();
+                stack.RemoveLast();
+                if (enumerator.MoveNext())
+                {
+                    var current = enumerator.Current;
+                    stack.Add(enumerator);
+
+                    if (current != null &&
+                        (descendIntoOperation == null || descendIntoOperation(current)))
+                    {
+                        if (!current.IsImplicit &&
+                            // This prevents non explicit operations like expression to be considered as ok
+                            (current.ConstantValue.HasValue || current.Type != null))
```

Comment:
This second part of the condition is copied from the `WithoutFullyImplicitOperations` helper but it is a bit more global than the other helper was suggesting (i.e. `or are explicit with no constant value or type, indicating it is not user written code`). The conditional operation linked to a `if` statement for example will be considered as not user written code _BUT_ the content of the condition will be properly analyzed.